pkg/egresspolicy/manager.go

```diff
@@ -287,14 +287,7 @@ func (manager *Manager) updateEgressMap(ips []string, config *Config) error {
 	for _, ip := range ips {
 		sip := net.ParseIP(ip).To4()
 		for _, dstCIDR := range config.dstCIDRs {
-			key := egressmap.NewKey(sip, dstCIDR.IP, dstCIDR.Mask)
-			value := &egressmap.EgressInfo4{}
-			// As currently designed, the egressIP serves two purposes, one for forwarding traffic
-			// to the gateway node, the other for SNATing the egress traffic on the gateway.
-			copy(value.TunnelEndpoint[:], config.egressIP)
-			copy(value.EgressIP[:], config.egressIP)
-
-			err := egressmap.EgressMap.Update(&key, value)
+			err := egressmap.InsertEgressGateway(sip, *dstCIDR, config.egressIP, config.egressIP)
```

Comment:
This doesn't seem super efficient as we'll make 1 `BPF_MAP_UPDATE_ELEM` call per gateway. We could introduce a `InsertEgressGateway` method to insert all gateways at the same time instead. That said, I'm not sure it really matters since this is only affecting the agent's speed and CPU consumption.

Reply:
I thought about this as well. No strong opinions here, but given that I'm not expecting the configuration to change that frequently, I initially found it hard to justify the extra code complexity/method to save a few `BPF_MAP_UPDATE_ELEM` calls (especially given that the maximum number of gateways per policy is currently 64).